version: "3"

services:
    db:
        image: mysql:8.0.33
        platform: linux/x86_64
        command: --default-authentication-plugin=mysql_native_password
        volumes:
            - db_app:/var/lib/mysql
        environment:
            MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
            MYSQL_DATABASE: ${MYSQL_DATABASE}
            MYSQL_USER: ${MYSQL_USER}
            MYSQL_PASSWORD: ${MYSQL_PASSWORD}
        ports:
            - 3306:3306
        networks:
            - rest_api_shop
    php:
        build:
            context: ./php
            args:
                TIMEZONE: ${TIMEZONE}
        volumes:
            - ../:/var/www/symfony:cached
            - ./php/conf.d/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
            - ./php/conf.d/error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
        networks:
            - rest_api_shop
    nginx:
        build:
            context: ./nginx
        volumes:
            - ../:/var/www/symfony:cached
        ports:
            - 8080:80
        networks:
            - rest_api_shop
        env_file:
            - .env.nginx.local
    rabbitmq:
        image: rabbitmq:3-management
        build:
            context: .
            dockerfile: rabbitmq/Dockerfile
        ports:
            - "15672:15672"
            - "5672:5672"
        healthcheck:
            test: [ "CMD", "curl", "-f", "http://localhost:15672" ]
            interval: 30s
            timeout: 10s
            retries: 5
        volumes:
            - rabbitmq-data:/var/lib/rabbitmq

volumes:
    rest_api_shop_var:
    rest_api_shop_vendor:
    db_app:
    rabbitmq-data:

networks:
    rest_api_shop:
